projects/openapi-io/src/parser/sourcemap.ts

```diff
@@ -52,6 +50,18 @@ export class JsonSchemaSourcemap {
     }
   }
 
+  logPointerInFile(
+    filePath: string,
+    sourcePointer: string,
+    targetPointer: string
+  ) {
```

Comment:
Rather than trying to reconstruct json pointers that included the path, I updated the sourcemap so we can pass in a file path.

Reply:
I think this will get weird if you the shared parameter was a ref. 

```
paths: 
  /xyz:
    parameters:
      - $ref: ./shared#parameters/auth
```

Reply:
I don't see any downside to doing it correctly (by reading from the sourcemap)

Reply:
It is reading the sourcemap https://github.com/opticdev/optic/pull/1566/files#diff-3d28014487c4e359675a91ff0ed1099359b36761c608fdfb91bfd7605b746a26R54-R65

and i've added it in as a test case here https://github.com/opticdev/optic/pull/1566/files#diff-c10d156f1637218487b466d310f1c89d779fd91f5f52accdd80b5b1422ec883eR24

Reply:
This is just to avoid having to do:
`filePath + jsonpointer` and then splitting that same file path again

Reply:
Sorry missed that. I see it now and this makes sense.